src/main/frontend/views/@layout.tsx

```diff
@@ -9,45 +9,51 @@ import {
   createMenuItems,
   useViewConfig,
 } from "@vaadin/hilla-file-router/runtime.js";
-import { Suspense, useEffect } from "react";
+import { Suspense } from "react";
 import { Outlet, useLocation, useNavigate } from "react-router-dom";
+import { Signal, signal, effect } from '@vaadin/hilla-react-signals';
 
-const defaultTitle = document.title;
+const vaadin = window.Vaadin as {
+  documentTitleSignal: Signal<string>;
+};
+vaadin.documentTitleSignal = signal("");
+effect(() =>  document.title = vaadin.documentTitleSignal.value);
 
 export default function MainLayout() {
-  const currentTitle = useViewConfig()?.title ?? defaultTitle;
+  const currentTitle = useViewConfig()?.title ?? '';
   const navigate = useNavigate();
   const location = useLocation();
 
-  useEffect(() => {
-    document.title = currentTitle;
-  }, [currentTitle]);
+  vaadin.documentTitleSignal.value = currentTitle;
```

Comment:
React strongly discourages rendering side-effects like this one. Please use an effect for state mutations to decouple them from rendering.

Reply:
Added `useEffect` for this.

Reply:
If this looks fine, I'll redo this to other similar PRs.